features/192-asset-gear/index.md

```diff
@@ -0,0 +1,640 @@
+      }
+    },
+    "required": ["headers"]
+}
+```
+
+- `exact_name`: The exact name of the asset.
+- `prefix`: If `exact_name` is not given, a random name is generated with `prefix` prepended.
+- `access`: The access control of the asset. `public` is the default.
+- `headers`: The HTTP headers of the asset when it is retrieved.
+- `headers.content-type`: The Content-Type header. `application/octet-stream` is the default.
+- `headers.content-disposition`: The Content-Disposition header.
+- `headers.content-encoding`: The Content-Encoding header. The developer must make sure the asset is compressed in the given compression.
+- `headers.content-length`: The Content-Length header. It is required.
+- `headers.content-md5`: The [Content-MD5](https://tools.ietf.org/html/rfc1864) header. Depending on the upload method, it may be ignored.
+- `headers.cache-control`: The Cache-Control header. If it is not given, a default one is added. To remove the header, set it to an empty string.
+- `headers.access-control-*`: The CORS headers. They are only useful if the developer needs to retrieve the asset with Fetch API.
+
+##### Request Example
+
+```json
+{
+  "headers": {
+    "content-type": "image/png"
+  }
+}
+```
+
+#### Response JSON Schema
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "asset_name": { "type": "string" },
+    "url": { "type": "string" },
+    "method": { "type": "string" },
+    "headers": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "name": { "type": "string" },
+          "value": { "type": "string" }
+        },
+        "required": ["name", "value"]
+      }
+    }
+  },
+  "required": ["asset_name", "url", "method", "headers"]
+}
+```
+
+- `asset_name`: The asset name. It becomes valid when the asset is uploaded.
+- `url`: The presigned URL to upload the asset.
+- `method`: The method to be used in the upload request.
+- `headers`: The headers to be included in the upload request.
+
+##### Response Example
+
+```json
+{
+  "asset_name": "myimage.png",
+  "url": "https://storage.skygearapps.com/bucket/myappid/myimage.png?...",
+  "method": "POST",
+  "headers": [
+    { "name": "content-type", "value": "image/png" }
+  ]
+}
+```
+
+#### Server Specification
+
+1. Let `ext` be an empty string.
+1. If `headers.content-type` is present, set `ext` to the file extension derived from it.
+1. If `exact_name` is given and non-empty, let `name` be `exact_name`. Set `headers.cache-control` to `no-cache` if it is absent.
+1. Otherwise let `name` to be the concatenation of `prefix`, a random string and `ext`. Set `headers.cache-control` to `max-age: 3600` if it is absent.
+1. Let `asset_id` be `<app-id>/<name>`.
+1. Remove any header in `headers` whose value is empty string.
+1. If `exact_name` is not given, ensure `asset_id` does not exist.
+1. Let `url` be the presigned URL.
+1. Return the presigned request.
+
+### POST /_asset/sign
```

Comment:
`sign` is a bit hard to understand, this endpoint is expected to be consumed by microservices as REST API? Maybe give it a more clear name like `get_signed_url`?

Reply:
Yes. Let me change it then.